src/components/idle-picker/IdlePicker.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react';
+import { default as RcSlider } from 'rc-slider';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Display from './display/Display';
+
+import styles from './IdlePicker.css';
+import 'rc-slider/assets/index.css';
+
+const marks = {
+    1: '01',
+    2: '',
+    3: '',
+    4: '',
+    5: '',
+    6: '',
+    7: '',
+    8: '',
+    9: '',
+    10: '10',
+};
+
+export default class IdlePicker extends Component {
+    static propTypes = {
+        defaultValue: PropTypes.number,
+        handlePickerChange: PropTypes.func,
+    };
+
+    static defaultProps = {
+        defaultValue: 3,
+    };
+
+    constructor(props) {
+        super(props);
+
+        let { defaultValue } = this.props;
+
+        if (defaultValue > 10 || defaultValue < 1) {
```

Comment:
Instead of doing this, can we use `Math.max(Math.min(defaultValue. 10), 3)` directly in the `this.state` definition?

Reply:
This means we no longer need the `let { defaultValue } = this.props;` as well because we access it just once.

Reply:
Good point, changed it 👍